### PR TITLE
DIV-6837: Update check your answers page for respondent solicitor details

### PR DESCRIPTION
**Pull request by madjava**
# Description

https://tools.hmcts.net/jira/browse/DIV-6837

CYA views for both Use Home Address and Solicitor Search have been updated to reflect the current updates introduced by the Respondent Solicitor Search feature. Correct data is now displayed for the `Where should your <xxx> divorce papers be sent?` and `Enter details of your <xxx> solicitor`. All updates to the Use home address CYA display view is behind a feature toggle as well.

Screenshot 2021-03-23 at 11 32 23

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/core/utils/respondentSolicitorSearchHelper.js [hunk 1]
```diff
@@ -258,8 +273,11 @@ module.exports = {
   errorsManualCleanup,
   parseManualAddress,
   parseAddressToManualAddress,
+  mapValidationErrors,
   mapRespondentSolicitorData,
+  mapRespondentSolicitorCyaData,
```
Comment (yarpo) [inline app/core/utils/respondentSolicitorSearchHelper.js:278]: if this is meant to be in alphabetical order C < D :)

One line up

File: app/core/utils/respondentSolicitorSearchHelper.js [hunk 2]
```diff
@@ -258,8 +273,11 @@ module.exports = {
   errorsManualCleanup,
   parseManualAddress,
   parseAddressToManualAddress,
+  mapValidationErrors,
```
Comment (yarpo) [inline app/core/utils/respondentSolicitorSearchHelper.js:276]: M is before P :)

File: app/core/utils/respondentSolicitorSearchHelper.js (hunk 1 above)
Comment (madjava) [reply app/core/utils/respondentSolicitorSearchHelper.js:278]: no not really i guess just a coincidence

**Review by yarpo — APPROVED** [review]

File: app/core/utils/respondentSolicitorSearchHelper.js [hunk 3]
```diff
@@ -162,8 +162,8 @@ const isManual = session => {
   return isEqual(session.searchType, 'manual');
 };
 
-const parseManualAddress = value => {
-  return value.split(/\r?\n/)
+const trimAndRemoveBlanks = list => {
+  return [].concat(list)
```
Comment (jacmurphy) [inline app/core/utils/respondentSolicitorSearchHelper.js:166]: ❤️

**Review by jacmurphy — APPROVED** [review]

File: app/steps/respondent/correspondence/solicitor-search/partials/checkYourAnswersTemplate.html [hunk 4]
```diff
@@ -3,9 +3,7 @@
     {{ content.question | safe }}
   </td>
   <td class="govuk-table__cell">
-    {% if session.respondentSolicitorAddress %}
-    <span class="address">{{ fields.cyaRespondentSolicitorAddress.value | safe }}</span>
-    {% endif %}
+    <span class="address">{{ fields.respondentSolicitorDisplayAddress.value | safe }}</span>
```
Comment (Dizrupto) [inline app/steps/respondent/correspondence/solicitor-search/partials/checkYourAnswersTemplate.html:6]: What would happen if there are no solicitors?

**Review by Dizrupto — APPROVED** [review]

File: app/steps/respondent/correspondence/solicitor-search/partials/checkYourAnswersTemplate.html (hunk 4 above)
Comment (madjava) [reply app/steps/respondent/correspondence/solicitor-search/partials/checkYourAnswersTemplate.html:6]: @Dizrupto, If manual entry, user has to provide data else can't submit the form, if search, address is provided by the api. If for any other unseen reason there is no `respondentSolicitorDisplayAddress` the template renders nothing if field does not exist. User will see nothing in screen.

**Pull request merged**